l10n_it_fatturapa_out/wizard/wizard_export_fatturapa.py

```diff
@@ -370,13 +370,14 @@ def _setDatiAnagraficiCessionario(self, partner, fatturapa):
                 raise UserError(
                     _('VAT number and fiscal code are not set for %s.') %
                     partner.name)
-        if partner.fiscalcode:
-            fatturapa.FatturaElettronicaHeader.CessionarioCommittente.\
-                DatiAnagrafici.CodiceFiscale = partner.fiscalcode
         if partner.vat:
             fatturapa.FatturaElettronicaHeader.CessionarioCommittente.\
                 DatiAnagrafici.IdFiscaleIVA = IdFiscaleType(
                     IdPaese=partner.vat[0:2], IdCodice=partner.vat[2:])
+        if partner.fiscalcode \
+                and (not partner.vat or partner.vat == partner.fiscalcode):
+            fatturapa.FatturaElettronicaHeader.CessionarioCommittente.\
+                DatiAnagrafici.CodiceFiscale = partner.fiscalcode
```

Comment:
Cioè non indichi mai il codice fiscale se diverso dalla partita IVA? Lo fai per evitare qualsiasi problema di controllo immagino, ma non dovrebbe superare il controllo ad es. inserendo il codice fiscale del titolare di un'azienda individuale?

Reply:
Ho controllato, supera il controllo se il codice fiscale è coerente con la partita IVA, non è che il problema sia quello (che ho visto) di inserire la P.IVA come IT... e il codice fiscale uguale IT...?

Reply:
In caso di gruppo d'impresa è obbligatorio il codice fiscale, come da specifiche:
`Nel caso di IdFiscaleIVA di gruppo IVA, il sistema verifica che sia presente anche il CodiceFiscale e che appartenga a uno dei partecipanti al gruppo; se il CodiceFiscale non è presente il file viene scartato con codice errore 00322; se è presente ma non è di partecipante al gruppo IVA, il file viene scartato con codice errore 00321.`

Reply:
@sergiocorato di preciso, quindi, quali dovrebbero essere i controlli? Così vedo di aggiornare la PR insieme ad @alessandrocamilli.

Reply:
Allora gestiamo i gruppi iva. Così gestiamo il caso specifico @sergiocorato?

Reply:
Beh, in sostanza sarebbe semplicemente da verificare che il codice fiscale sia plausibile (ma c'è mica un modulo installato che lo fa?) perchè va messo in caso di gruppo iva (e non è vietato in tutti gli altri).
La fattura che si è bloccata che codice fiscale riportava?

Reply:
Da quanto leggo [qui](https://github.com/ema/pycodicefiscale) basta usare la funzione `isvalid(code) -> bool` per verificare se un codice fiscale è sintatticamente corretto

Reply:
@sergiocorato quindi come proporresti di modificare?
Questo potrebbe andare?
```
        if partner.vat:
            fatturapa.FatturaElettronicaHeader.CessionarioCommittente.\
                DatiAnagrafici.IdFiscaleIVA = IdFiscaleType(
                    IdPaese=partner.vat[0:2], IdCodice=partner.vat[2:])
        if partner.fiscalcode and isvalid(partner.fiscalcode):
            fatturapa.FatturaElettronicaHeader.CessionarioCommittente.\
                DatiAnagrafici.CodiceFiscale = partner.fiscalcode
```
Perché nelle specifiche SDI c'è scritto che devono essere coerenti i valori di `<IdFiscaleIVA>` e `<CodiceFiscale>`, ma non è specificato cosa significhi "essere coerenti"...

Reply:
@sergiocorato abbiamo aggiornamenti qui?